### PR TITLE
Add back 7zip support, and add support for lzma, xz and bzip2 tarballs

### LINKED ISSUE #164: Support .tar.gz, .tar.bz2 and .tar.xz archives
Please add support for .tar.gz, .tar.bz2 and .tar.xz archives. Also .tar (non-compressed one) seems to be supported, but _View_ is disabled, I can only _Extract_ it. Please also enable _View_ (for both .tar and compressed variants like .tar.gz, .tar.bz2 and .tar.xz).

### PR DESCRIPTION
**Pull request by TranceLove**
Add back 7zip support (#1170) which was removed by #1183 in master, and add support of lzma, xz and bzip2 tarballs to fix #164. Unit tests with archives included.

RarDecompressor was reporting incorrect decompressed size (not sure why - if it's the `rar` on my machine or `junrar` itself), but since nobody complained about this (so far), its test case was omitted for the time being.

### CODE REVIEW

**Comment by VishalNehra** [comment]
Works great! However shows empty fragment when viewing a non supported compression type. Happens on current master as well so can't blame this PR. 
Might as well create a separate issue for it where we show some form of snackbar to tell user about it.

File: app/src/main/java/com/amaze/filemanager/filesystem/compressed/extractcontents/helpers/SevenZipExtractor.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+        if (entry.isDirectory()) {
+            FileUtil.mkdir(new File(outputDir, name), context);
+            return;
+        }
+        File outputFile = new File(outputDir, name);
+        if (!outputFile.getParentFile().exists()) {
+            FileUtil.mkdir(outputFile.getParentFile(), context);
+        }
+        // Log.i("Amaze", "Extracting: " + entry);
```
Comment (EmmanuelMess) [inline app/src/main/java/com/amaze/filemanager/filesystem/compressed/extractcontents/helpers/SevenZipExtractor.java]: Commented code.

**Comment by EmmanuelMess** [comment]
I will add a "do not merge" tag for you to remove when you want another round of tests for this PR.

**Comment by VishalNehra** [comment]
@TranceLove text files opens just fine for me. Which specific compressed tarballs don't open for you?

**Comment by TranceLove** [comment]
>@TranceLove text files opens just fine for me. Which specific compressed tarballs don't open for you?

My test was on Oneplus One running Slim7 (7.1.2) using a [A Protocol Buffers release tarball](https://github.com/google/protobuf/releases/download/v3.0.0/protobuf-cpp-3.0.0.tar.gz); also repackaged to produce tar.xz and tar.lzma.

At my test, all three tar.gz, tar.lzma and tar.xz had the symptom that @EmmanuelMess described:
1. Open an archive
2. Randomly choose a text file (.txt)
3. Tap on the text file
4. Toast prompting user "Please wait"
5. Nothing happened afterwards (expected was TextEditorActivity or prompt to open with Amaze Text Editor)

Try compare with and without 210835a.

**Comment by TranceLove** [comment]
This had gone too far behind. Will start it over in a new PR.

**Pull request closed without merging**